test/iterable_matchers_test.dart

```diff
@@ -167,14 +176,14 @@ void main() {
         unorderedMatches([3, 1]),
         "Expected: matches [<3>, <1>] unordered "
         "Actual: [1, 2] "
-        "Which: has no match for <3> at index 0");
+        "Which: has no match for <3>");
```

Comment:
fwiw, the index was actually useful here previously, especially in the case of objects that don't have good string representations. Imagine a `List<Foo>` where `Foo` has no custom `toString` method (or a not useful one), how do I know what one didn't match?

Reply:
I think I can add back the index. Do we think it's helpful to list out multiple matchers that were unmatched or should I stop at the first one like the old implementation?

Reply:
Added back the index and shortened the rest of the message to "along with n other unmatched"